### PR TITLE
feat(Token): Migrate to CSS modules behind feature flag Pt 2

### PR DESCRIPTION
**Pull request by randall-krauskopf**
Related To: https://github.com/github/primer/issues/4367

This PR is part 2 of breaking apart a [larger PR](https://github.com/primer/react/pull/5237) into grokkable parts.

Apologies for the larger PR but there was an odd interplay with the specificity across the various components that made it hard to break this PR into smaller chunks

### CODE REVIEW

**Comment by randall-krauskopf** [comment]
Integration tests are failing due to the regression that should be fixed soon.

File: packages/react/src/Token/AvatarToken.module.css [hunk 1]
```diff
@@ -0,0 +1,37 @@
+:root {
+  --spacing: calc(var(--base-size-4) * 2);
+}
+
+.AvatarContainer {
+  display: block;
+}
+
+/* TODO: Remove this once the avatar component is converted to CSS modules */
```
Comment (jonrohan) [inline packages/react/src/Token/AvatarToken.module.css:9]: Avatar is converted https://github.com/primer/react/pull/5221

File: packages/react/src/Token/AvatarToken.module.css [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+.Avatar {
+  width: 100% !important;
+  height: 100% !important;
+}
+
+.Token {
+  padding-left: var(--base-size-4) !important;
+}
+
+.AvatarContainer[data-size='small'] {
```
Comment (jonrohan) [inline packages/react/src/Token/AvatarToken.module.css:19]:
```suggestion
.AvatarContainer:where([data-size='small']) {
```

File: packages/react/src/Token/AvatarToken.module.css [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  width: calc(16px - var(--spacing));
+  height: calc(16px - var(--spacing));
+}
+
+.AvatarContainer[data-size='medium'] {
```
Comment (jonrohan) [inline packages/react/src/Token/AvatarToken.module.css:24]:
```suggestion
.AvatarContainer:where([data-size='medium']) {
```

File: packages/react/src/Token/AvatarToken.module.css (hunk 2 above)
Comment (randall-krauskopf) [reply packages/react/src/Token/AvatarToken.module.css:19]: Hi @jonrohan, I'll get these changes in. Quick question though, why is `:where` preferred over the bare attribute selector?

Comment (jonrohan) [reply packages/react/src/Token/AvatarToken.module.css:19]: Where will reduce the specificity on the attribute selector and make it easier to override the styles in dotcom. We try and add it whenever the attribute lands on the same dom node where we add the passed in className

**Review by jonrohan — APPROVED** [review]

**Review by hussam-i-am — APPROVED** [review]

**Pull request merged**